pkg/cache/cache.go

```diff
@@ -414,6 +419,18 @@ func combineTransform(inherited, current toolscache.TransformFunc) toolscache.Tr
 }
 
 func defaultOpts(config *rest.Config, opts Options) (Options, error) {
+	logger := log.WithName("setup")
+
+	// Use the rest HTTP client for the provided config if unset
+	if opts.HTTPClient == nil {
+		var err error
+		opts.HTTPClient, err = rest.HTTPClientFor(config)
+		if err != nil {
+			logger.Error(err, "Failed to get HTTP client")
+			return opts, fmt.Errorf("could not create HTTP client from config")
```

Comment:
return the original error

Reply:
@alvaroaleman @vincepri this comment was posted just a bit too late, I think it is a valid remark. Do I create a new PR to fix this issue?

Reply:
Yup just create a follow-up PR that's fine

Reply:
see https://github.com/kubernetes-sigs/controller-runtime/pull/2161